code/game/objects/structures/mineral_doors.dm

```diff
@@ -118,6 +120,30 @@
 		if(do_after(user, 40 * digTool.toolspeed * hardness, target = src) && src)
 			to_chat(user, "<span class='notice'>You finished digging.</span>")
 			deconstruct(TRUE)
+	else if(istype(W, /obj/item/stack/sheet/wood) && user.a_intent == INTENT_HELP)
+		var/obj/item/stack/sheet/wood/S = W
+		if(!density)
+			to_chat(user, "<span class='warning'>\The [src] must be closed!</span>")
+			return
+		if(S.get_amount() < 2)
+			to_chat(user, "<span class='warning'>You need at least 2 planks of wood to barricade this!</span>")
+			return
+		if(door_barricaded)
+			to_chat(user, "<span class='warning'>There's already a barricade here!</span>")
+			return
+		if((/mob) in get_turf(src))
+			to_chat(user, "<span class='warning'>There's someone blocking [src]!</span>")
+			return
+		to_chat(user, "<span class='notice'>You start barricading [src]...</span>")
+		if(do_after_once(user, 4 SECONDS, target = src) && S.get_amount() > 2)
+			S.use(2)
+			var/obj/structure/barricade/wooden/crude/newbarricade = new(loc)
+			transfer_fingerprints_to(newbarricade)
+			to_chat(user, "<span class='notice'>You barricade [src] shut.</span>")
+			user.visible_message("<span class='notice'>[user] barricades [src] shut.</span>")
+			return
+		else if(S.get_amount() < 2)
+			to_chat(user, "<span class='warning'>You ran out of wood during construction.</span>")
```

Comment:
This sadly doesn't guard against anything, as it'll only be triggered if you break the `do_after()`, which won't be broken purely by having less than 2 planks